noxfile.py

```diff
@@ -49,54 +49,33 @@
-def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> None:
-    """Run pytest benchmarks for selected icon4py model subpackages."""
+def benchmark_model(session: nox.Session) -> None:
+    """Run pytest benchmarks."""
     _install_session_venv(session, extras=["dace", "io", "testing"], groups=["test"])
 
-    results_json_path = os.path.abspath(f"results_{subpackage.replace('/', '_')}.json")
-    with session.chdir(f"model/{subpackage}"):
-        session.run(
-            *f"pytest \
-            -v \
-            --benchmark-only \
-            --benchmark-warmup=on \
-            --benchmark-warmup-iterations=30 \
-            --benchmark-json={results_json_path}".split(),
-            *session.posargs,
-            success_codes=[0, NO_TESTS_COLLECTED_EXIT_CODE],
-        )
+    session.run(
+        *f"pytest \
+        -v \
+        --benchmark-only \
+        --benchmark-warmup=on \
+        --benchmark-warmup-iterations=30 \
+        --benchmark-json=pytest_benchmark_results_{session.python}.json \
+        ./model".split(),
+        *session.posargs,
+    )
 
-def merge_pytest_benchmark_results(bencher_json_file_name: str) -> None:
-    """Gather all benchmark results files and merge them into a single file."""
-    merged_results = {"benchmarks": []}
-    files = glob.glob("results_*.json")
-    for file in files:
-        try:
-            with open(file, "r") as f:
-                data = json.load(f)
-                merged_results["benchmarks"].extend(data["benchmarks"])
-                # preserve the first file's metadata for a valid pytest-benchmark json file
-                for key in data.keys():
-                    if key != "benchmarks" and key not in merged_results:
-                        merged_results[key] = data[key]
-        except:
-            # Empty file, i.e. no benchmarks
-            continue
-    with open(bencher_json_file_name, "w") as f:
-        json.dump(merged_results, f, indent=4)
-
-@nox.session(python=["3.10", "3.11"],
-             requires=["benchmark_model-{python}" + f"({subpackage.id})" for subpackage in MODEL_SUBPACKAGE_PATHS])
+@nox.session(python=["3.10", "3.11"], requires=["benchmark_model-{python}"])
 def bencher_baseline(session: nox.Session) -> None:
-    """Run pytest benchmarks and upload them using Bencher (https://bencher.dev/) (cloud or self-hosted)."""
-    bencher_json_file_name = f"merged_benchmark_results_{session.python}.json"
-    merge_pytest_benchmark_results(bencher_json_file_name)
-
+    """
+    Run pytest benchmarks and upload them using Bencher (https://bencher.dev/) (cloud or self-hosted).
+    This session is used only on the main branch to create the historical baseline.
+    The historical baseline is used to compare the performance of the code in the PRs.
+    """
     session.run(
         *f"bencher run \
         --project {os.environ['BENCHER_PROJECT']} \
         --token {os.environ['BENCHER_API_TOKEN']} \
```

Comment:
I've noticed here that `nox` output in the CI leaks the token of the self hosted bencher. Probably we should find a way to keep this secret

Reply:
I found a way to protect the secret env vars -> https://github.com/C2SM/icon4py/blob/506a183dc935033edbb6dba6db3ddde28229ab8f/noxfile.py#L83

Bencher uses these env vars if they exist. So, if we do not pass them explicitly in the bencher run command, then they are not exposed and bencher uses the ones that are already exported in the env.

Reply:
https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/5125340235196978/2255149825504674/-/jobs/9415342313#L1680